src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -366,7 +366,7 @@ trait Infer extends Checkable {
       // optimize type variables wrt to the implicit formals only; ignore the result type.
       // See test pos/jesper.scala
       def variance = restpe match {
-        case mt: MethodType if mt.isImplicit && isFullyDefined(pt) => MethodType(mt.params, AnyTpe)
+        case mt: MethodType if mt.isImplicit && isFullyDefined(pt) => MethodType(mt.params, AnyTpe) //
```

Comment:
LGTM, we'll ignore all implicit param lists, rather than the only one, when approximating the type used to calculate the variance (and direction of solving) of the type params